src/back/TicketManager/src/Core/TicketManager.Core.Domain/Events/Event.cs

```diff
@@ -23,5 +24,66 @@ public Event(Guid organizerId, string name, string description, string location,
         Description = description;
         Location = location;
         Date = date;
+        Status = EventStatus.Verified;
+    }
+
+    public void ChangeEventStatus(EventStatus status)
+    {
+        switch(status)
+        {
+            case EventStatus.Verified:
+                if(Status != EventStatus.Unverified &&  Status != EventStatus.Cancelled)
+                {
+                    throw new InvalidOperationException("Cannot change event status to Verified from current status.");
+                }
+                break;
+            case EventStatus.Published:
+                if(Status != EventStatus.Verified)
+                {
+                    throw new InvalidOperationException("Cannot change event status to Published from current status.");
+                }
+                break;
+            case EventStatus.Opened:
+                if (Status != EventStatus.Published && Status != EventStatus.Closed)
+                {
+                    throw new InvalidOperationException("Cannot change event status to Opened from current status.");
+                }
+                break;
+            case EventStatus.Closed:
+                if (Status != EventStatus.Opened)
+                {
+                    throw new InvalidOperationException("Cannot change event status to Closed from current status.");
+                }
+                break;
+            case EventStatus.Finished:
+                if (Status != EventStatus.Opened && Status != EventStatus.Closed)
```

Comment:
Open and close finished event?

Reply:
If we want to change `EventStatus` to `Finished` it has to be in `Opened` or `Closed` state first
![image](https://user-images.githubusercontent.com/43820607/236765185-ee638fd5-15e1-4170-8ed8-8a6cb9eff422.png)